html/browsers/the-window-object/named-access-on-the-window-object/navigated-named-objects.window.js

```diff
@@ -0,0 +1,67 @@
+// META: script=/common/get-host-info.sub.js
+
+function echoURL(content) {
+  return `/common/echo.py?content=${encodeURIComponent(content)}`;
+}
+
+function setSrc(frame, type, content) {
+  if (type === "same-origin") {
+    frame.src = echoURL(content);
+  } else if (type === "cross-site") {
+    frame.src = `${get_host_info().HTTP_NOTSAMESITE_ORIGIN}${echoURL(content)}`;
+  } else {
+    frame.srcdoc = content;
+  }
+}
+
+["srcdoc", "same-origin", "cross-site"].forEach(type => {
+  const initialType = type === "srcdoc" ? type : "same-origin";
+
+  [
+    {
+      "namedObject": "<div id=abc></div>",
+      "namedObjectLocalName": "div"
+    },
+    {
+      "namedObject": "<object name=abc></object>",
+      "namedObjectLocalName": "object"
+    },
+    {
+      "namedObject": "<iframe id=abc></iframe>",
+      "namedObjectLocalName": "iframe"
+    }
+  ].forEach(testData => {
+    async_test(t => {
+      const frame = document.createElement("iframe");
+      t.add_cleanup(() => frame.remove());
+      setSrc(frame, initialType, `<script>function f() { return abc }</script>${testData.namedObject}`);
+      frame.onload = t.step_func(() => {
+        const f = frame.contentWindow.f,
```

Comment:
It is very surprising that Chrome/Safari fail with "f is not a function". f should always be a function... Maybe this is one of those cases where we get multiple onloads, and you need to skip the about:blank?

Reply:
I thought about that too and alerted window.location.href at this spot and it didn't give me about:blank.

Reply:
Hah, I think I figured it out, I'm hitting some kind of non-standard XSS prevention thing.